File: golang/tests/conformance_test.go
package main_test

import (
	"path/filepath"
	"runtime"
	"testing"

	"github.com/jkassis/courier/pkg/conformance"
)

func TestConformanceFixture(t *testing.T) {
	_, here, _, ok := runtime.Caller(0)
	if !ok {
		t.Fatal("runtime.Caller failed")
	}
	fixturePath := filepath.Join(filepath.Dir(here), "..", "..", "conformance", "fixtures", "suite.v1.json")
	if err := conformance.RunFixture(fixturePath); err != nil {
		t.Fatalf("conformance fixture failed: %v", err)
	}
}

File: typescript/tests/conformance.test.ts
import path from 'node:path'
import { runConformanceFixture } from '../src/pkg/conformance'

describe('conformance fixture', () => {
  it('passes shared suite.v1 scenarios', () => {
    const fixturePath = path.resolve(__dirname, '..', '..', 'conformance', 'fixtures', 'suite.v1.json')
    runConformanceFixture(fixturePath)
  })
})
